"unicodedecodeerror (once again) with format() but not with concatenation i have a class chunk with text fields title and text. when i want to print them, i get (surprise, surprise!) unicodedecodeerror. it gives me an error when i try to format an output string, but when i just concatenate text and title and return it, i get no error:..class chunk:.  # init, fields, .....  # this implementation will give me an error.  def __str__( self ):.    return u'{0} {1}'.format ( enc(self.text), enc(self.title) )..  # but this is ok - all is printed without error.  def __str__( self ):.    return enc(self.text) + enc(self.title)..def enc(x):.  return x.encode('utf-8','ignore') # tried many combinations of arguments......c = chunk().c.text, c.title = ... # feed from external file.print c...bum! error!..return u'{0} {1}'.format ( enc(self.text), enc(self.title) ).unicodedecodeerror: 'ascii' codec can't decode byte 0xc2 in position 2844: ordinal not in range(128)...i think i used all the possible combinations of encode/decode/utf-8/ascii/replace/ignore/.....(the blank unicode issue is really irritating!)"
